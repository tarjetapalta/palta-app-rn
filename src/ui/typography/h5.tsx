import React, { FunctionComponent } from 'react';

import { Text, TextProps } from './base';

export const H5: FunctionComponent<TextProps> = ({
  /** Test id for test */
  testID = 'H5',
  /** Text */
  children,
  /** Text color */
  color,
  /** Text font weight */
  weight,
  /** Text font opacity */
  emphasis,
  /** Text align */
  textAlign,
}) => (
  <Text
    testID={testID}
    color={color}
    size="s"
    weight={weight}
    emphasis={emphasis}
    textAlign={textAlign}
  >
    {children}
  </Text>
);
