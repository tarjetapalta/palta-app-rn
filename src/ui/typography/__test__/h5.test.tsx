import React from 'react';
import renderer from 'react-test-renderer';

import WithTheme from '../../../theme/context';
import { H5 } from '../h5';

describe('h5 component', () => {
  it('basic H5 with color "text", emphasis "high", weight "medium" and testID "H5"', () => {
    const result = renderer
      .create(
        <WithTheme>
          <H5>Some text</H5>
        </WithTheme>,
      )
      .toJSON();

    expect(result).toMatchSnapshot();
  });
  it('basic H5 with color "primary", emphasis "medium"', () => {
    const result = renderer
      .create(
        <WithTheme>
          <H5 color="primary" emphasis="medium">
            Some text
          </H5>
        </WithTheme>,
      )
      .toJSON();

    expect(result).toMatchSnapshot();
  });
  it('basic with custom testId', () => {
    const result = renderer
      .create(
        <WithTheme>
          <H5 testID="custom testID">Some text</H5>
        </WithTheme>,
      )
      .toJSON();

    expect(result).toMatchSnapshot();
  });
  it('basic H5 with weight "light"', () => {
    const result = renderer
      .create(
        <WithTheme>
          <H5 weight="regular">Some text</H5>
        </WithTheme>,
      )
      .toJSON();

    expect(result).toMatchSnapshot();
  });
});
