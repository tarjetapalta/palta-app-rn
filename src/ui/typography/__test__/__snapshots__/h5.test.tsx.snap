// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`h5 component basic H5 with color "primary", emphasis "medium" 1`] = `
<Text
  color="primary"
  emphasis="medium"
  numberOfLines={0}
  size="s"
  style={
    Array [
      Object {
        "color": "#91b086",
        "fontFamily": "BwModelica-Regular",
        "fontSize": 14,
        "opacity": 0.6,
        "textAlign": "center",
      },
    ]
  }
  testID="H5"
>
  Some text
</Text>
`;

exports[`h5 component basic H5 with color "text", emphasis "high", weight "medium" and testID "H5" 1`] = `
<Text
  numberOfLines={0}
  size="s"
  style={
    Array [
      Object {
        "color": "#283238",
        "fontFamily": "BwModelica-Regular",
        "fontSize": 14,
        "opacity": 0.87,
        "textAlign": "center",
      },
    ]
  }
  testID="H5"
>
  Some text
</Text>
`;

exports[`h5 component basic H5 with weight "light" 1`] = `
<Text
  numberOfLines={0}
  size="s"
  style={
    Array [
      Object {
        "color": "#283238",
        "fontFamily": "BwModelica-Regular",
        "fontSize": 14,
        "opacity": 0.87,
        "textAlign": "center",
      },
    ]
  }
  testID="H5"
  weight="regular"
>
  Some text
</Text>
`;

exports[`h5 component basic with custom testId 1`] = `
<Text
  numberOfLines={0}
  size="s"
  style={
    Array [
      Object {
        "color": "#283238",
        "fontFamily": "BwModelica-Regular",
        "fontSize": 14,
        "opacity": 0.87,
        "textAlign": "center",
      },
    ]
  }
  testID="custom testID"
>
  Some text
</Text>
`;
